plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/deploy/CloudSdkDeployProjectHandler.java

```diff
@@ -0,0 +1,171 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package com.google.cloud.tools.eclipse.appengine.localserver.deploy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.swt.widgets.DirectoryDialog;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.console.MessageConsole;
+import org.eclipse.wst.server.core.IRuntime;
+
+import com.google.cloud.tools.eclipse.appengine.localserver.Activator;
+import com.google.cloud.tools.eclipse.appengine.localserver.CloudSdkUtils;
+import com.google.cloud.tools.eclipse.appengine.localserver.GCloudCommandDelegate;
+import com.google.cloud.tools.eclipse.util.ActiveProjectFinder;
+import com.google.cloud.tools.eclipse.util.MessageConsoleUtilities;
+import com.google.cloud.tools.eclipse.util.ProcessUtilities;
+
+/**
+ * Handler for the Cloud SDK deploy action.
+ */
+public class CloudSdkDeployProjectHandler extends AbstractHandler {
+  public static final String TITLE = "Cloud SDK Deploy";
+  private static final String WAR_SRC_DIR_DEFAULT = "src/main/webapp";
+
+  @Override
+  public Object execute(ExecutionEvent event) throws ExecutionException {
+    // Get initial project selection
+    final IProject project = ActiveProjectFinder.getSelectedProject(event);
+    if (project == null) {
+      Activator.logAndDisplayError(null, TITLE, "Cannot find selected project");
+      return null;
+    }
+
+    final IRuntime runtime;
```

Comment:
try combining these try blocks so the variable declarations aren't split. Short try blocks are a common anti-pattern.